compiler/src/java_plugin/cpp/java_generator.h

```diff
@@ -52,7 +52,7 @@ string ServiceClassName(const google::protobuf::ServiceDescriptor* service);
 // Writes the generated service interface into the given ZeroCopyOutputStream
 void GenerateService(const google::protobuf::ServiceDescriptor* service,
                      google::protobuf::io::ZeroCopyOutputStream* out,
-                     ProtoFlavor flavor);
+                     ProtoFlavor flavor, bool disable_version);
```

Comment:
I think I'd opt for a struct before a map and especially before a vector. I'm not too concerned yet with having a lot of arguments to that method, since there is only one caller of it.